src/platform_specific/affinity.win.cc

```diff
@@ -1,22 +1,22 @@
-#pragma once
+#ifdef _WIN32
 
 #include <Windows.h>
 #include <cassert.h>
 #include <io.h>
 
+#include "affinity.h"
+#include "utils.h"
+
 typedef DWORD_PTR native_cpu_set;
 
 uint32_t affinity_cores_available() {
 	native_cpu_set affinity_base_mask;
 	native_cpu_set sys_affinity_mask; // not really needed
 	assert(GetProcessAffinityMask(GetCurrentProcess(), &affinity_base_mask, &sys_affinity_mask) == 0 && "Error retrieving base affinity mask.");
 
-	int bit_mask_length = sizeof(native_cpu_set) * 8;
-	native_cpu_set power = 1;
-	uint32_t count = 0;
-	for(uint32_t i = 0; i < bit_mask_length; i++) {
-		if((power & affinity_base_mask) != 0) { ++count; }
-		power = power << 1;
-	}
+	static uint32_t count = celerity::details::popcount(affinity_base_mask);
```

Comment:
Not really a bug here, but the `affinity_base_mask` will still be computed each time the function is called, even if the `count` that's returned is cached. So this should probably also be wrapped in a helper function/lambda.

Reply:
I argued (offline) in favor of not doing any caching at all, since we call this function exactly *once*. Should it ever end up on some hot path (which I seriously doubt), we can optimize it then. Either way, `count` shouldn't be `static`.